bundles/org.openhab.binding.meater/src/main/java/org/openhab/binding/meater/internal/api/MeaterRestAPI.java

```diff
@@ -0,0 +1,177 @@
+                    for (Device meaterProbe : devices) {
+                        meaterProbeThings.put(meaterProbe.id, meaterProbe);
+                    }
+                } else {
+                    meaterProbeThings.clear();
+                }
+            }
+            return true;
+        } catch (MeaterException | JsonSyntaxException e) {
+            logger.warn("Failed to refresh! {}", e.getMessage());
+        }
+        return false;
+    }
+
+    private void login() throws MeaterException {
+        try {
+            // Login
+            String json = "{ \"email\": \"" + configuration.email + "\",  \"password\": \"" + configuration.password
+                    + "\" }";
+            Request request = httpClient.newRequest(API_ENDPOINT + LOGIN).method(HttpMethod.POST);
+            request.header(HttpHeader.ACCEPT, JSON_CONTENT_TYPE);
+            request.header(HttpHeader.CONTENT_TYPE, JSON_CONTENT_TYPE);
+            request.content(new StringContentProvider(json), JSON_CONTENT_TYPE);
+
+            logger.trace("HTTP POST Request {}.", request.toString());
+
+            ContentResponse httpResponse = request.send();
+            if (httpResponse.getStatus() != HttpStatus.OK_200) {
+                throw new MeaterException("Failed to login " + httpResponse.getContentAsString());
+            }
+            // Fetch JWT
+            json = httpResponse.getContentAsString();
+            JsonObject jsonObject = JsonParser.parseString(json).getAsJsonObject();
+            JsonObject childObject = jsonObject.getAsJsonObject("data");
+            this.authToken = childObject.get("token").getAsString();
+            this.userId = childObject.get("userId").getAsString();
+        } catch (TimeoutException | ExecutionException | JsonParseException e) {
+            throw new MeaterException(e);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new MeaterException(e);
+        }
+    }
+
+    private String getFromApi(String uri) throws MeaterException {
+        try {
+            for (int i = 0; i < MAX_RETRIES; i++) {
+                try {
+                    Request request = httpClient.newRequest(API_ENDPOINT + uri).method(HttpMethod.GET);
+                    request.header(HttpHeader.AUTHORIZATION, "Bearer " + authToken);
+                    request.header(HttpHeader.ACCEPT, JSON_CONTENT_TYPE);
+                    request.header(HttpHeader.CONTENT_TYPE, JSON_CONTENT_TYPE);
+                    request.header(HttpHeader.ACCEPT_LANGUAGE, localeProvider.getLocale().getLanguage());
+
+                    ContentResponse response = request.send();
+                    String content = response.getContentAsString();
+                    logger.trace("API response: {}", content);
+
+                    if (response.getStatus() == HttpStatus.UNAUTHORIZED_401) {
+                        logger.debug("getFromApi failed, HTTP status: {}", HttpStatus.UNAUTHORIZED_401);
```

Comment:
Small optimization possible:
```suggestion
                        logger.debug("getFromApi failed, HTTP status: 401");
```

Reply:
Fixed!